crates/aranya-quic-channels/tests/test.rs

```diff
@@ -0,0 +1,137 @@
+        if let Some(len) = channel1.recv_stream(target.as_mut_slice()).await {
+            assert_eq!(&target[..len], b"hello2");
+        } else {
+            panic!("no data received");
+        }
+
+        // Test sending messages
+        channel1.send_message(&Bytes::from("message1")).await?;
+        tokio::time::sleep(Duration::from_millis(100)).await;
+        if let Some(len) = channel2.recv_message(target.as_mut_slice()).await {
+            assert_eq!(&target[..len], b"message1");
+        } else {
+            panic!("no data received");
+        }
+        channel2.send_stream(&Bytes::from("message2")).await?;
+        tokio::time::sleep(Duration::from_millis(100)).await;
+        if let Some(len) = channel1.recv_stream(target.as_mut_slice()).await {
+            assert_eq!(&target[..len], b"message2");
+        } else {
+            panic!("no data received");
+        }
+        let big_data = {
+            let mut rng = rand::thread_rng();
+            let mut data = vec![0u8; 1024 * 1024 * 3 / 2];
+            rand::Rng::fill(&mut rng, &mut data[..]);
+            Bytes::from(data)
+        };
+        channel1.send_message(&big_data).await?;
+        tokio::time::sleep(Duration::from_millis(100)).await;
+        // Send message should return the complete message
+        if let Some(len) = channel2.recv_message(target.as_mut_slice()).await {
+            assert_eq!(&target[..len], &big_data[..]);
+        } else {
+            panic!("no data received");
+        }
+
+        channel1.send_stream(&big_data).await?;
+        tokio::time::sleep(Duration::from_millis(100)).await;
+        let mut total_len: usize = 0;
+        let mut total_pieces: i32 = 0;
+        // Send stream should return the message in pieces
+        while let Some(len) = channel2.recv_stream(&mut target[total_len..]).await {
+            total_pieces = total_pieces.checked_add(1).expect("Pieces overflow");
+            total_len = total_len.checked_add(len).expect("Length overflow");
+            if total_len >= big_data.len() {
+                break;
+            }
+            tokio::time::sleep(Duration::from_millis(20)).await;
+        }
+        assert!(total_pieces > 1);
+        assert_eq!(total_len, big_data.len());
+        assert_eq!(&target[..total_len], &big_data[..]);
+    } else {
+        panic!("channel is not available");
+    }
+
+    Ok(())
+}
+
+fn get_server(cert: String, key: String) -> Result<Server> {
+    let server = Server::builder()
+        .with_tls((&cert[..], &key[..]))?
+        .with_io("127.0.0.1:0")?
+        .with_congestion_controller(Bbr::default())?
+        .start()?;
+    Ok(server)
+}
+
+fn spawn_channel_listener(
+    aqc_client: Arc<TMutex<AqcClient>>,
+    server: Server,
+) -> Result<SocketAddr> {
+    let server_addr = server.local_addr()?;
+    tokio::spawn(run_channels(aqc_client, server));
+    Ok(server_addr)
+}
+
+fn make_client() -> Arc<TMutex<ClientState<TestEngine, MemStorageProvider>>> {
```

Comment:
Our codebase generally uses `new` in the method name when creating new objects.
```suggestion
fn new_client() -> Arc<TMutex<ClientState<TestEngine, MemStorageProvider>>> {
```